tests/fpga/hbm_bank_split_test.py

```diff
@@ -0,0 +1,55 @@
+import dace
+from hbm_copy_fpga_test import mkc
+from dace.dtypes import StorageType
+from dace.transformation.dataflow import HbmBankSplit
+from dace.transformation import optimizer
+import numpy as np
+
+
+def test_even_split_3d():
+    sdfg = dace.SDFG("hbm_bank_split_even_split_3d")
+    s, b, a = mkc(sdfg, None, "b", "a", StorageType.CPU_Heap,
+                  StorageType.CPU_Heap, [8, 50, 50, 50], [100, 100, 100], "b")
+    for xform in optimizer.Optimizer(sdfg).get_pattern_matches(
+            patterns=HbmBankSplit):
+        xform.apply(sdfg)
+    b = np.random.uniform(0, 100, [8, 50, 50, 50]).astype(np.int32)
+    sdfg(a=a, b=b)
+    assert np.allclose(a[0:50, 0:50, 0:50], b[0, :, :, :])
+    assert np.allclose(a[50:100, 50:100, 50:100], b[7, :, :, :])
+    assert np.allclose(a[0:50, 50:100, 0:50], b[2, :, :, :])
+
+
+def test_second_dim_split_2d():
+    sdfg = dace.SDFG("hbm_bank_split_sec_dim_split2d")
+    s, a, b = mkc(sdfg, None, "a", "b", StorageType.CPU_Heap,
+                  StorageType.CPU_Heap, [10, 100], [10, 10, 10], "b")
+    for xform in optimizer.Optimizer(sdfg).get_pattern_matches(
+            patterns=HbmBankSplit):
+        xform.split_array_info = [1, 10]
+        xform.apply(sdfg)
+    a = np.random.uniform(0, 10, [10, 100]).astype(np.int32)
+    sdfg(a=a, b=b)
+    for i in range(10):
+        assert np.allclose(a[0:10, 10 * i:(10 * i + 10)], b[i])
+
+
+def test_explicit_split_3d():
+    sdfg = dace.SDFG("hbm_bank_split_explicit_3d")
+    s, a, b = mkc(sdfg, None, "a", "b", StorageType.CPU_Heap,
+                  StorageType.CPU_Heap, [120, 100, 100], [24, 40, 50, 25])
+    for xform in optimizer.Optimizer(sdfg).get_pattern_matches(
+            patterns=HbmBankSplit):
+        xform.split_array_info = [3, 2, 4]
+        xform.apply(sdfg)
+    a = np.random.uniform(0, 100, [120, 100, 100]).astype(np.int32)
+    sdfg(a=a, b=b)
+    assert np.allclose(a[80:120, 50:100, 75:100], b[23])
+    assert np.allclose(a[0:40, 50:100, 75:100], b[7])
+    assert np.allclose(a[40:80, 0:50, 25:50], b[9])
+
+
+if __name__ == "__main__":
```

Comment:
I'm missing an example of why this transformation is useful in practice, it's not so easy to understand just from looking at it. Or maybe it needs to be explained in the docstring why you need this

Reply:
The actual use case is this: Suppose you implement say e.g. dot to use HBM, hence you distribute both inputs across many banks, and your inputs (on device) are maybe of shape [16, N / 16]. But likely you would like to initialize this from just 2 1D arrays of shape [N]. Now here this transformation gets helpful, because you can simply connect the 2 arrays by an edge and apply - the host array will be distributed automatically.

Then there are situations, e.g when computing A^T*x where it's also usefull to not split along dimension 0. Here for example if A is split along rows (the same column is always on 1 bank) one could avoid reducing over multiple banks, so it would make sense to split along dimension 1 in that case. (It's because if A is split along columns you would actually get multiple y vectors which you have to add up to get the final result for A^T*x)